src/map/mob.cpp

```diff
@@ -2475,8 +2475,14 @@ void mob_damage(struct mob_data *md, struct block_list *src, int damage)
 		int i;
 		for(i = 0; i < DAMAGELOG_SIZE; i++){ // must show hp bar to all char who already hit the mob.
 			struct map_session_data *sd = map_charid2sd(md->dmglog[i].id);
-			if( sd && check_distance_bl(&md->bl, &sd->bl, AREA_SIZE) ) // check if in range
-				clif_monster_hp_bar(md, sd->fd);
+			if( sd && check_distance_bl(&md->bl, &sd->bl, AREA_SIZE) ) { // check if in range
```

Comment:
Needs additional call above the loop to update md->master.
Master will never be inside damage log and also needs HP update.